front/app/modules/commercial/user_custom_fields/admin/containers/CustomFields/RegistrationCustomFieldEdit/RegistrationCustomFieldSettings.tsx

```diff
@@ -66,18 +62,23 @@ class RegistrationCustomFieldSettings extends React.Component<Props, State> {
       });
   };
 
-  renderFn = (props) =>
-    this.props.customField && (
-      <RegistrationCustomFieldForm
-        {...props}
-        mode="edit"
-        customFieldId={this.props.customField.id}
-        builtInField={isBuiltInField(this.props.customField)}
-      />
+  renderFn = (props) => {
+    const { customField } = useUserCustomFieldOutletContext();
+
+    return (
+      customField && (
+        <RegistrationCustomFieldForm
+          {...props}
+          mode="edit"
+          customFieldId={customField.id}
+          builtInField={isBuiltInField(customField)}
+        />
+      )
     );
+  };
 
   render() {
-    const { customField } = this.props;
+    const { customField } = useUserCustomFieldOutletContext();
```

Comment:
Same remark here, we can do the data loading the component it seems. No need for the outletContext I think.

Reply:
See comment above, the hook won't work here and I'm not sure the best solution (without rewriting the component or creating a new data fetch function, though likely one of those is necessary)

Reply:
I'll look at it!